File: Example_VI_CtCI_Python/Example12.py
def permutation(str, prefix):
    if (len(str) == 0):
        print(prefix)
    else:
        i = 0
        while (i < len(str)):
            rem = str[0:i] + str[i + 1:]
            permutation(rem, prefix + str[i])
            i += 1

str = "Lucy"
permutation(str, "")

File: Examples_VI_CtCI_C/Example 12/Example12.c
#include <stdio.h>
#include <string.h>
// O (n2 * n!) runtime. //not finished.

void permutation (char str[], char prefix[]) {
	int i;
	
	if (strlen(str) == 0) {
		printf("%s\n", prefix);
	} else {
		for (i = 0; i < strlen(str); i++) {
			char rem[] = strstr(0, i) + strstr(i + 1);
			permutation(rem, prefix + str.at(i));
		}
	}
}

int main() {
	char str[]= "Lucy";
	char prefix[] = "";
	permutation(str, prefix);
	
	return 0;
}
